service/src/main/java/bio/terra/catalog/service/CatalogStatusService.java

```diff
@@ -1,26 +1,33 @@
 package bio.terra.catalog.service;
 
 import bio.terra.catalog.config.StatusCheckConfiguration;
+import bio.terra.catalog.datarepo.DatarepoService;
+import bio.terra.catalog.iam.SamService;
 import bio.terra.catalog.model.SystemStatusSystems;
 import java.sql.Connection;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
-import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
 
-@Component
+@Service
 public class CatalogStatusService extends BaseStatusService {
   private static final Logger logger = LoggerFactory.getLogger(CatalogStatusService.class);
 
   private final NamedParameterJdbcTemplate jdbcTemplate;
 
   @Autowired
   public CatalogStatusService(
-      NamedParameterJdbcTemplate jdbcTemplate, StatusCheckConfiguration configuration) {
+      NamedParameterJdbcTemplate jdbcTemplate,
+      StatusCheckConfiguration configuration,
+      SamService samService,
+      DatarepoService datarepoService) {
     super(configuration);
     this.jdbcTemplate = jdbcTemplate;
-    super.registerStatusCheck("CloudSQL", this::databaseStatus);
+    registerStatusCheck("CloudSQL", this::databaseStatus);
+    registerStatusCheck("SAM", samService::status);
+    registerStatusCheck("Data Repo", datarepoService::status);
```

Comment:
This looks really nice

Reply:
I agree! I copied it from https://github.com/DataBiosphere/terra-billing-profile-manager/blob/rt-toaz-44/service/src/main/java/bio/terra/profile/service/status/ProfileStatusService.java